guides-markdown/events.md

```diff
@@ -0,0 +1,115 @@
+| `tx` | [TX](https://github.com/bcoin-org/bcoin/blob/master/lib/primitives/tx.js), [CoinView](https://github.com/bcoin-org/bcoin/blob/master/lib/coins/coinview.js) | [Mempool](https://github.com/bcoin-org/bcoin/blob/master/lib/mempool/mempool.js): `addEntry()` | _mempool_&#8594;[Pool](https://github.com/bcoin-org/bcoin/blob/master/lib/net/pool.js) (returns&nbsp;TX&nbsp;only)<br> _mempool_&#8594;_pool_&#8594;[SPVNode](https://github.com/bcoin-org/bcoin/blob/master/lib/node/spvnode.js) (returns&nbsp;TX&nbsp;only)<br> _mempool_&#8594;[FullNode](https://github.com/bcoin-org/bcoin/blob/master/lib/node/fullnode.js) (returns&nbsp;TX&nbsp;only)<br>_SPVNode_,&nbsp;_FullNode_&#8594;[NodeClient](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/nodeclient.js) (returns&nbsp;TX&nbsp;only) |
+| `tx` | [TX](https://github.com/bcoin-org/bcoin/blob/master/lib/primitives/tx.js)| [Pool](https://github.com/bcoin-org/bcoin/blob/master/lib/net/pool.js): `_handleTX()`<br>(only if there is no mempool, i.e. SPV) | _pool_&#8594;[SPVNode](https://github.com/bcoin-org/bcoin/blob/master/lib/node/spvnode.js) |
+| `tx` | [TX](https://github.com/bcoin-org/bcoin/blob/master/lib/primitives/tx.js), [Details](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/txdb.js) | [TXDB](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/txdb.js): `insert()` | _txdb_&#8594;[WalletDB](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/walletdb.js) (also&nbsp;returns&nbsp;Wallet)<br>_txdb_&#8594;[Wallet](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/wallet.js)|
+| `double spend` | [MempoolEntry](https://github.com/bcoin-org/bcoin/blob/master/lib/mempool/mempoolentry.js) | [Mempool](https://github.com/bcoin-org/bcoin/blob/master/lib/mempool/mempool.js): `removeDoubleSpends()` | |
+| `balance` | [Balance](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/txdb.js) | [TXDB](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/txdb.js): `insert()`, `confirm()`, `disconnect()`, `erase()` | _txdb_&#8594;[WalletDB](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/walletdb.js) (also&nbsp;returns&nbsp;Wallet)<br>_txdb_&#8594;[Wallet](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/wallet.js)|
+| `address` |  [[WalletKey](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/walletkey.js)] | [Wallet](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/wallet.js): `_add()`,  |  _wallet_&#8594;[WalletDB](https://github.com/bcoin-org/bcoin/blob/master/lib/wallet/walletdb.js) (returns&nbsp;parent&nbsp;Wallet,&nbsp;[WalletKey]) |
+
+## Socket Events
+
+Websocket connections in bcoin are handled by two servers, one for `Node` and one for `Wallet`.
+Those servers each have child objects such as `Chain`, `Mempool`, `Pool`, and `WalletDB`, and
+relay events from them out the socket. To receive an event, the socket client must watch a
+channel (such as `chain`, `mempool`, or `auth`) or join the a wallet
```

Comment:
This is definitely something that would be good to have a guide or code snippet for at some point, either in this doc or separately and linked to here.

Reply:
Cool, gunna add examples in the next commit

Reply:
Awesome